pkg/cluster/generateconfig.go

```diff
@@ -28,143 +24,67 @@ import (
+	return ocDynamicValidator.Dynamic(ctx)
+}
 
-	if m.doc.OpenShiftCluster.Properties.Install == nil {
-		// we don't re-call Dynamic on subsequent entries here.  One reason is
-		// that we would re-check quota *after* we had deployed our VMs, and
-		// could fail with a false positive.
-		timeoutCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
-		defer cancel()
-		wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
-			err = m.ocDynamicValidator.Dynamic(ctx)
-			if azureerrors.HasAuthorizationFailedError(err) ||
-				azureerrors.HasLinkedAuthorizationFailedError(err) {
-				m.log.Print(err)
-				return false, nil
-			}
-			return err == nil, err
-		}, timeoutCtx.Done())
-		if err != nil {
-			return err
-		}
-	}
+func (m *manager) generateInstallConfig(ctx context.Context) (*installconfig.InstallConfig, *installconfig.PlatformCreds, *releaseimage.Image, error) {
 
 	resourceGroup := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
 
-	if m.env.DeploymentMode() != deployment.Development {
-		rp := m.acrtoken.GetRegistryProfile(m.doc.OpenShiftCluster)
-		if rp == nil {
-			// 1. choose a name and establish the intent to create a token with
-			// that name
-			rp = m.acrtoken.NewRegistryProfile(m.doc.OpenShiftCluster)
-
-			m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
-				m.acrtoken.PutRegistryProfile(doc.OpenShiftCluster, rp)
-				return nil
-			})
-			if err != nil {
-				return err
-			}
-		}
-
-		if rp.Password == "" {
-			// 2. ensure a token with the chosen name exists, generate a
-			// password for it and store it in the database
-			password, err := m.acrtoken.EnsureTokenAndPassword(ctx, rp)
-			if err != nil {
-				return err
-			}
-
-			rp.Password = api.SecureString(password)
-
-			m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
-				m.acrtoken.PutRegistryProfile(doc.OpenShiftCluster, rp)
-				return nil
-			})
-			if err != nil {
-				return err
-			}
-		}
-	}
-
-	m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
-		if doc.OpenShiftCluster.Properties.SSHKey == nil {
-			sshKey, err := rsa.GenerateKey(rand.Reader, 2048)
-			if err != nil {
-				return err
-			}
-
-			doc.OpenShiftCluster.Properties.SSHKey = x509.MarshalPKCS1PrivateKey(sshKey)
-		}
-
-		if doc.OpenShiftCluster.Properties.StorageSuffix == "" {
-			doc.OpenShiftCluster.Properties.StorageSuffix, err = randomLowerCaseAlphanumericStringWithNoVowels(5)
-			if err != nil {
-				return err
-			}
-		}
-
-		return nil
-	})
-	if err != nil {
-		return err
-	}
-
 	pullSecret, err := pullsecret.Build(m.doc.OpenShiftCluster, string(m.doc.OpenShiftCluster.Properties.ClusterProfile.PullSecret))
 	if err != nil {
-		return err
+		return nil, nil, nil, err
```

Comment:
ugly as hell but I'm +1 on it as an interim step.

Reply:
we can get rid of at least 1 of these with 4.6, at least :)